src/Sarif.Multitool/Rules/StepValuesMustFormOneBasedSequence.cs

```diff
@@ -33,46 +38,46 @@ protected override IEnumerable<string> FormatIds
             }
         }
 
-        protected override void Analyze(CodeFlow codeFlow, string codeFlowPointer)
+        protected override void Analyze(ThreadFlow threadFlow, string threadFlowPointer)
         {
-            var pointer = new JsonPointer(codeFlowPointer);
-            JToken codeFlowToken = pointer.Evaluate(Context.InputLogToken);
+            var pointer = new JsonPointer(threadFlowPointer);
+            JToken threadFlowToken = pointer.Evaluate(Context.InputLogToken);
 
-            JProperty locationsProperty = codeFlowToken.Children<JProperty>()
+            JProperty locationsProperty = threadFlowToken.Children<JProperty>()
                 .FirstOrDefault(prop => prop.Name.Equals(SarifPropertyName.Locations, StringComparison.Ordinal));
             if (locationsProperty != null)
             {
-                JArray annotatedCodeLocationArray = locationsProperty.Value as JArray;
-                string annotatedCodeLocationsPointer = codeFlowPointer.AtProperty(SarifPropertyName.Locations);
+                JArray threadFlowLocationsArray = locationsProperty.Value as JArray;
+                string threadFlowLocationsPointer = threadFlowPointer.AtProperty(SarifPropertyName.Locations);
 
                 ReportMissingStepProperty(
-                    annotatedCodeLocationArray,
-                    annotatedCodeLocationsPointer);
+                    threadFlowLocationsArray,
+                    threadFlowLocationsPointer);
 
                 ReportInvalidStepValues(
-                    codeFlow.Locations.ToArray(),
-                    annotatedCodeLocationArray,
-                    annotatedCodeLocationsPointer);
+                    threadFlow.Locations.ToArray(),
+                    threadFlowLocationsArray,
+                    threadFlowLocationsPointer);
             }
         }
 
         private void ReportInvalidStepValues(
-            AnnotatedCodeLocation[] locations,
-            JArray annotatedCodeLocationArray,
-            string annotatedCodeLocationsPointer)
+            ThreadFlowLocation[] locations,
+            JArray threadFlowLocationsArray,
+            string threadFlowLocationsPointer)
         {
-            JObject[] annotatedCodeLocationObjects = annotatedCodeLocationArray.Children<JObject>().ToArray();
+            JObject[] threadFlowLocationObjects = threadFlowLocationsArray.Children<JObject>().ToArray();
 
             for (int i = 0; i < locations.Length; ++i)
             {
                 // Only report "invalid step value" for locations that actually specify
```

Comment:
>Only report "invalid step value" for locations that actually specify [](start = 19, length = 68)

gee,all this makes me realize we have a step priority that you must populate but which can always be inferred simply by tracking the index value of the object array.

if a producer doesn't provide it, any consumer that wanted it could manufacture it. any consumer could correct the bad output from the tool. makes me think we should just cut this property and put a note in the spec: it may be helpful to compute a 'step' value for display. this can be inferred from the array index, etc..

Reply:
Nice idea. It does assume a "smart" consumer, and not two people poring over a raw log file together.